apps/web/src/app/(private)/evaluations/(evaluation)/[evaluationUuid]/dashboard/_components/ConnectedDocumentsTable/index.tsx

```diff
@@ -0,0 +1,115 @@
+'use client'
+
+import { useMemo } from 'react'
+
+import { HEAD_COMMIT } from '@latitude-data/core/browser'
+import type { ConnectedDocumentWithMetadata } from '@latitude-data/core/repositories'
+import {
+  Skeleton,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+  Text,
+} from '@latitude-data/web-ui'
+import { formatCostInMillicents } from '$/app/_lib/formatCostInMillicents'
+import { useNavigate } from '$/hooks/useNavigate'
+import { ROUTES } from '$/services/routes'
+import useProjects from '$/stores/projects'
+
+const ConnectedDocumentTableRow = ({
+  document,
+  onSelect,
+}: {
+  document: ConnectedDocumentWithMetadata
+  onSelect: () => void
+}) => {
+  const { data: projects, isLoading: isProjectsLoading } = useProjects()
+  const projectName = useMemo(() => {
+    if (isProjectsLoading) return null
+
+    return projects?.find((project) => project.id === document.projectId)?.name
+  }, [document.projectId, isProjectsLoading, projects])
+
+  const promptPath = useMemo(() => {
+    return document.path.split('/').slice(0, -1).join('/')
+  }, [document.path])
+
+  const promptName = useMemo(() => {
+    return document.path.split('/').pop()
+  }, [document.path])
+
+  return (
+    <TableRow
+      key={document.documentUuid}
+      className='cursor-pointer border-b-[0.5px] h-12 max-h-12 border-border'
+      onClick={onSelect}
+    >
+      <TableCell className='nowrap'>
+        <Text.H4 color='foregroundMuted' noWrap>
+          {promptPath}
+          {Boolean(promptPath) && '/'}
+          <Text.H4 noWrap>{promptName}</Text.H4>
```

Comment:
I would truncate this name after a certain number of characters, the preview you shared doesn't look very good with very long path names 😅